lib/battle/CBattleInfoCallback.cpp

```diff
@@ -855,6 +855,28 @@ AccessibilityInfo CBattleInfoCallback::getAccesibility() const
 		ret[BattleHex(0, y)] = EAccessibility::SIDE_COLUMN;
 	}
 
+	//special battlefields with logically unavailable tiles
+	std::vector<BattleHex> impassableHexes;
+	switch(battleGetBattlefieldType().num)
+	{

```

Comment:
After all... I think it is pretty unlikely we will need to handle more cases. Will change to "if" soon